syntax = "proto3";

option java_multiple_files = true;
option java_package = "com.betasolutions.grpc.movie";
option java_outer_classname = "MovieProto";
option objc_class_prefix = "RTG";

package movie;


service MovieLibrary {

    // Bidirectional streaming rpc to allow user to get and add movies to collection
    rpc getAndUpdateMovieLibrary (stream Movie) returns (stream Movie) {
    }

    // Client side streaming rpc to allow user to fetch movies from the server
    rpc listMovies (ListMoviesRequest) returns (stream Movie) {
    }

}
message Movie {
    int64 id = 1;
    string movie_name = 2;
    string genre = 4;
    string client_id = 5;
}

message ListMoviesRequest {
    int64 current_id = 1;
    int32 max_page_size = 2;
    string genre = 3;
    string client_id = 4;
}